### DISCUSSION TITLE
Time gap between the lookback period and the forecast period

### DISCUSSION THREAD

**Discussion by EdxDv** [Nixtla/neuralforecast#470] [Q&A] [answered]
Hi everyone ! we are interested in forecasting H days ahead after a gap of p days, actually what we want to forecast is y[t+p+1:t+p+H](we do not need to forecast y[t+1:t+p]), do you have any ideas on how to adjust the models ? Otherwise, do you think it's a good idea to forecast the total period and just ignore the first p forecasts ?

**Comment by kdgutier** [comment, marked as answer]
Hey @EdxDv, 

- The simplest solution is to continue with a normal NeuralForecast model and filter the predictions to only keep them after the first `p` horizon steps.
- If you want to explore a simple model customization an interesting possibility would be to hack the horizon mask in [the training_step method](https://github.com/Nixtla/neuralforecast/blob/main/neuralforecast/common/_base_recurrent.py#L339) to make the loss function to only depend on the observations after `p`. This second solution can be achieved simply by overwriting a NeuralForecast model `train_step`.

Let us know your thoughts.

**Reply by EdxDv** [reply]
Thank you so much for your suggestions, we really appreciate your fast reply !

**Reply by EdxDv** [reply]
Sorry for bothering you again, do you think modifying the outsample_mask in the training_step method before feeding it to the loss function, namely by setting outsample_mask[:,0:p,:]=0 in the BaseRecurrent class and outsample_mask[:,0:p]=0 in the BaseWindows class will do the trick ?

**Reply by kdgutier** [reply]
Yes, `outsample_mask[:,p:,:]=0`=>`[0,0,0,1,1,1,1,1]` would specialize the network on the horizon after the first p steps.
It would be interesting to validate that such strategy helps the accuracy of the later steps.
